### PR TITLE
Convert remoteconfig subcommand to an app

### PR DESCRIPTION
**Pull request by djmitche**
### What does this PR do?

Converts the `agent remote-config` command to an Fx App.

### Motivation

Top-down approach to componentization.

### Additional Notes

This PR is against a base branch containing #13699 and #13662.  Once those are merged, I'll change the base of this PR to `main` and rebase, but it can be reviewed in the interim.

### Describe how to test/QA your changes

For each of 7.40.0 and a build of the main branch, do the following, and verify that the output is the same.

Configure an agent with remote config enabled (DD_REMOTE_CONFIG_ENABLED=true) and get a shell adjacent to it.  Run `agent remote-config` and see something like
```
Fetching the configuration and director repos state..
Error: Couldn't get the repositories state: rpc error: code = Unknown desc = remote configuration service not initialized
```

Run `agent remote-config show` and see the same output.  In no case should that output be colored.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: cmd/agent/subcommands/remoteconfig/command.go [hunk 1]
```diff
@@ -14,42 +14,53 @@ import (
...
-				return fmt.Errorf("Remote configuration is not enabled")
-			}
-			return state(cmd, args)
+			return fxutil.OneShot(state,
+				fx.Supply(cliParams),
+				fx.Supply(core.BundleParams{
+					ConfFilePath:      globalParams.ConfFilePath,
+					ConfigLoadSecrets: false,
+				}),
+				core.Bundle,
```
Comment (hush-hush) [inline cmd/agent/subcommands/remoteconfig/command.go:51]: Question: the function takes the `config` and then the `cliParams`. Does the order we supply arguments matters to `fx.New(opts...)`  ? I thought it did not.

If not could we invert the order to match the argument order of the function, I think it makes the code more intuitive to understand.

**Review by hush-hush — APPROVED** [review]
Added a question, looks good otherwise.

File: cmd/agent/subcommands/remoteconfig/command.go [hunk 2]
```diff
@@ -64,7 +75,7 @@ func state(cmd *cobra.Command, args []string, dialOpts ...grpc.DialOption) error
 	}
 	ctx = metadata.NewOutgoingContext(ctx, md)
 
-	cli, err := agentgrpc.GetDDAgentSecureClient(ctx, dialOpts...)
+	cli, err := agentgrpc.GetDDAgentSecureClient(ctx)
```
Comment (ogaca-dd) [inline cmd/agent/subcommands/remoteconfig/command.go:78]: Why this change?

**Review by ogaca-dd — CHANGES_REQUESTED** [review]

File: cmd/agent/subcommands/remoteconfig/command.go (hunk 1 above)
Comment (djmitche) [reply cmd/agent/subcommands/remoteconfig/command.go:51]: The order does not matter, you're right.

I'm not sure what you mean by "the function", though?

File: cmd/agent/subcommands/remoteconfig/command.go (hunk 2 above)
Comment (djmitche) [reply cmd/agent/subcommands/remoteconfig/command.go:78]: This slice is always empty (`state` is only called in one place), so I just removed it rather than build a way to pass it with Fx.

File: cmd/agent/subcommands/remoteconfig/command.go (hunk 1 above)
Comment (djmitche) [reply cmd/agent/subcommands/remoteconfig/command.go:51]: Oh, the one-shot function.  I'll just change the order of its arguments.

**Review by ogaca-dd — APPROVED** [review]

**Review by ameske — APPROVED** [review]
